tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -1277,10 +1289,19 @@ static void SetDefaultConfigArguments(ClientContext &context) {
 	auto &config = ClientConfig::GetConfig(context);
 	config.enable_progress_bar = true;
 
-	if (DuckDBPyConnection::IsJupyter()) {
-		// Set the function used to create the display for the progress bar
-		context.config.display_create_func = JupyterProgressBarDisplay::Create;
+	if (!DuckDBPyConnection::IsJupyter()) {
+		return;
 	}
+	if (!HasJupyterProgressBarDependencies()) {
+		// Disable progress bar altogether
+		config.system_progress_bar_disable_reason = "either 'ipywidgets' or 'IPython' modules are not installed, so we "
```

Comment:
We don't have a way to test Jupyter behavior currently, which is also why bugs like this creep in :/

Reply:
That would simulate a CLI interactive shell, no?

Reply:
I'm not sure if the suggested change makes sense, since we need both these packages.
The wording in your suggestion makes it sound like we only need one of the two.

Reply:
> That would simulate a CLI interactive shell, no?

Ah, I was under the incorrect assumption that you could use ipywidgets on the command line as well. My bad!

Reply:
Much better yes, thankyou 😅